provider/k8s/k8s.go

```diff
@@ -240,13 +240,27 @@ func (p *Provider) heartbeat() error {
 		return errors.WithStack(err)
 	}
 
+	spotCnt := 0
+	onDemandCnt := 0
+
+	for i := range ns.Items {
+		switch ns.Items[i].Annotations["eks.amazonaws.com/capacityType"] {
```

Comment:
@nightfury1204, each engine (aws, gcp, etc) as your own "Heartbeat" function that track fields specific for the provider. As sport and on-demand are related to aws provider, I think you should include this logic inside of the aws file. `/provider/aws/heartbeat.go`